test/Microsoft.TestPlatform.AcceptanceTests/Extension/NetCoreTargetFrameworkDataSource.cs

```diff
@@ -23,24 +23,54 @@ public class NetCoreTargetFrameworkDataSource : Attribute, ITestDataSource
         /// </summary>
         /// <param name="useDesktopRunner">To run tests with desktop runner(vstest.console.exe)</param>
         /// <param name="useCoreRunner">To run tests with core runner(dotnet vstest.console.dll)</param>
-        public NetCoreTargetFrameworkDataSource(bool useDesktopRunner = true, bool useCoreRunner = true)
+        public NetCoreTargetFrameworkDataSource(
+            bool useDesktopRunner = true, 
+            // adding another runner is not necessary until we need to start building against another 
+            // sdk, because the netcoreapp2.1 executable is forward compatible
+            bool useCoreRunner = true,
+            bool useNetCore21Target = true, 
+            // laying the ground work here for tests to be able to run against 3.1 but not enabling it for
+            // all tests to avoid changing all acceptance tests right now
+            bool useNetCore31Target = false)
         {
             this.dataRows = new List<object[]>(6);
 
             if (useDesktopRunner)
             {
-                this.dataRows.Add(new object[]
+                if (useNetCore21Target)
```

Comment:
It seems there's a lot of code duplication below. How about using something like this:

```
private void AddRunner(useTarget, runnerFramework, targetFramework) {
if (!useTarget) return;

this.dataRows.Add(new object[]
{
    new RunnerInfo(runnerFramework, targetFramework)
});
}
```

Then you can have statements like:
```
AddRunner(useNetCore21Target, IntegrationTestBase.DesktopRunnerFramework, AcceptanceTestBase.Core21TargetFramework);
AddRunner(useNetCore31Target, IntegrationTestBase.DesktopRunnerFramework, AcceptanceTestBase.Core31TargetFramework);
...
```

Reply:
That's not a very obvious design. The method is called Add* so it should Add. Not TryAdd (which has a different connotation), or AddIfEnabled. If anything I would refactor to:

```cs
if (useNetCore21Target) 
{
    AddRunnerDataRow(... 
}
```